documentation/reference/ci_cd.md

```diff
@@ -0,0 +1,491 @@
+# CI + CD workflow
+
+The CI + CD workflow is the primary pipeline of the project. It runs the tests
+for all layers of the stack (which have changed in the current PR or commit)
+and, if the tests pass, publishes the documentation to the docs site and deploys
+the API and frontend to their respective staging ECS targets.
+
+It's a complex workflow with many jobs that depend on each other and the number
+of paths it can take becomes huge because jobs can be skipped based on the
+contents of a PR or commit.
+
+## Output
+
+The CI + CD workflow has four main outputs when new commits are pushed to the
+`main` branch, although not all outputs are expected on each commit. The first
+of these, "documentation" is also applicable to PRs.
+
+- **Documentation:** For each PR which affects the documentation, the
+  documentation is built and deployed to a preview site. When a new commit is
+  pushed to `main` that affects the documentation, the documentation is built
+  and deployed to the [root of the docs site](https://docs.openverse.org).
+
+  See job [`emit-docs`](#emit-docs) for details.
+
+- **Docker images:** When a new commit is pushed to `main`, the Docker images
+  for each of the affected services is pushed to
+  [GHCR](https://github.com/orgs/WordPress/packages?repo_name=openverse). These
+  images are tagged with the SHA of the commit.
+
+  See job [`publish-images`](#publish-images).
+
+- **Frontend staging:** When a new commit is pushed to `main` which affects the
+  frontend service and the tests for the frontend pass, a new
+  [staging deployment of the frontend](https://staging.openverse.org/) is
+  triggered. The deployment uses the image published above.
+
+  See job [`deploy-frontend`](#deploy-frontend).
+
+- **API staging:** When a new commit is pushed to `main` which affects the API
+  service and the tests for the API pass, a new
+  [staging deployment of the API](https://api-staging.openverse.engineering/) is
+  triggered. The deployment uses the image published above.
+
+  See job [`deploy-api`](#deploy-api).
+
+## Actions
+
+The workflow depends on four actions that have been taken out of the workflow
+for reuse and readability.
+
+### `get-changes`
```

Comment:
They're all under a H2 of "Actions".

Reply:
The sidebar also shows the hierarchy pretty clearly imo.

<img width="177" alt="Screenshot 2023-04-18 at 6 03 23 PM" src="https://user-images.githubusercontent.com/16580576/232802020-7defab49-ad1a-4390-9ff7-9a61c6aa499d.png">

Reply:
Sure, but unfortunately you could get linked to `get-changes` and easily confuse it if you didn't check the sidebar. I often view the documentation in a half-screen window which hides the sidebar as well. I don't think relying on ambient context to disambiguate two things named exactly the same thing is reliable.

Additionally, because they share the same header name, the generated anchor for the second is meaningless: https://docs.openverse.org/_preview/1001/reference/ci_cd.html#id1

Reply:
Disregard my previous comment, the changes in the latest commit fix the issue :+1: